controlplane/pkg/sidecars/nsm-init.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2018-2019 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sidecars
+
+import (
+	"context"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/networkservicemesh/networkservicemesh/sdk/client"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+)
+
+// NSMClientApp - an initial client application used by nsm-init side-car
+type NSMClientApp interface {
```

Comment:
I like this structure very much... makes the whole thing a library you can simply drop into a main() :)